solutions/instances/main.tf

```diff
@@ -151,7 +152,25 @@ module "kms" {
 # COS
 #######################################################################################################################
 
-module "cos" {
+# workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
+resource "time_sleep" "wait_for_authorization_policy" {
+  depends_on      = [ibm_iam_authorization_policy.policy]
+  count           = var.skip_cos_kms_auth_policy ? 0 : 1
+  create_duration = "30s"
+}
+
+# Create IAM Authorization Policy to allow COS to access KMS for the encryption key
+resource "ibm_iam_authorization_policy" "policy" {
+  count                       = var.skip_cos_kms_auth_policy ? 0 : 1
```

Comment:
this count needs to be smarter. Your not going to want to create an auth policy if user is passing existing buckets